### PR TITLE
feat: Update README with code examples

### PR DESCRIPTION
**Pull request by Edu-B**
## Description
This pull request adds examples for file upload and authentication functionality. The new examples demonstrate how to upload files to the application and implement authentication.

## Changes 
- Add code examples for `HMAC`, `API_KEY` and `OAuth` authorization
- Add code examples for both upload methods
- Add Alfred brief description

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,60 +1,123 @@
...
 ```
 
-### Usage
+## Usage
 
-1. **Initialize the Client**
+### Initialize the Client
 
-   Begin by creating an instance of the Alfred client using your preferred authentication method:
+   Begin by creating an instance of the Alfred client using your preferred authentication method along with the desired environment type (Production or Staging). The following examples demonstrate how to initialize the client with different authentication methods:
```
Comment (rroa) [inline README.md:47]: what are the differences between environments?

File: README.md [hunk 2]
```diff
@@ -1,60 +1,123 @@
-# TagShelf.Alfred.ApiWrapper
+# Overview
 
 The `TagShelf.Alfred.ApiWrapper` is a comprehensive .NET library designed to facilitate seamless interactions with the Alfred API. It's tailored to support a wide range of .NET applications, from modern .NET Core and .NET Standard projects to legacy .NET Framework 4.7.2 applications, providing a robust, strongly-typed interface for efficient development.
 
 It provides a robust, strongly-typed interface for efficient development, with specialized domains for handling Tags, Jobs, Files, Deferred Sessions, and Data Points.
 
+## Alfred
+
+Alfred is a powerful document processing platform that enables you to extract, index, and search through large document collections with ease. It offers a wide range of features, including:
```
Comment (rroa) [inline README.md:9]: This is great, but the features are not arranged in a way that clearly or accurately reflects the process you want to undertake with the library.

File: README.md [hunk 3]
```diff
@@ -1,60 +1,123 @@
...
    // Upload remote file
    var uploadResult = await alfred.File.UploadAsync(new FileUploadRequest
    {
-       Urls = new List<string> { "http://example.com/file.pdf" },
+       Urls = new List<string> { "https://pdfobject.com/pdf/sample.pdf" },
        Metadata = "{'DocumentType':'Invoice'}",       
    });   
    ```
 
+#### Upload File from Stream
```
Comment (rroa) [inline README.md:101]: Why would you choose one or the other?

**Review by rroa — CHANGES_REQUESTED** [review]
These updates are positive and moving in the right direction. However, we need to concentrate on the main tasks you carry out with Alfred and consider how a third party would use it. It's important to distinguish between uploading deferred files and uploading remote files. We have not yet described the process of uploading files (whether with deferred sessions or not), creating a job, monitoring its status, and then retrieving the results after processing.

File: README.md [hunk 4]
```diff
@@ -11,50 +11,139 @@ It provides a robust, strongly-typed interface for efficient development, with s
...
-   With the client initialized, you're ready to perform API operations. Access the File and Job domains as follows:
+1. Initialize the Alfred client.
+2. Choose the method to upload the file (from URL or stream).
+3. Create a request object with the necessary parameters.
+4. Call the appropriate method to upload the file.
+5. Handle the response accordingly.(The response will contain the file ID or the job ID, depending on the method used.)
+
+#### Upload File (From URL or URLs)
+
+This method allows you to upload a file from a URL or multiple URLs to Alfred. You can also specify metadata for the file during the upload process you have to take into account that this method will automatically trigger the job processing.
```
Comment (ypk46) [inline README.md:83]:
```suggestion
This method enables you to upload a file from one or multiple URLs to Alfred. During the upload process, you can also specify metadata for the file. Please note that using this method will automatically trigger job processing.
```

File: README.md [hunk 5]
```diff
@@ -11,50 +11,139 @@ It provides a robust, strongly-typed interface for efficient development, with s
...
    {
-       Urls = new List<string> { "http://example.com/file.pdf" },
+       Urls = new List<string> { "https://pdfobject.com/pdf/sample.pdf" },
        Metadata = "{'DocumentType':'Invoice'}",       
    });   
    ```
 
+#### Upload File from Stream
+
+This method allows you to upload a file from a stream to Alfred and associate it with a session ID this method will not trigger the job processing this method is ideal for uploading files that are part of a larger job.
```
Comment (ypk46) [inline README.md:111]:
```suggestion
This method enables you to upload a file from a stream to Alfred and associate it with a specific session ID. Unlike other methods, this one does not trigger job processing, making it ideal for uploading files that are part of a larger job.
```

File: README.md [hunk 6]
```diff
@@ -11,50 +11,139 @@ It provides a robust, strongly-typed interface for efficient development, with s
...
+
+- **Extraction**: Can extract specific data from PDFs, images, and other documents with ease using its powerful extraction engine.
+
+- **Tagging**: Tag documents based on their content, making it easy to organize and search through large document collections.
+
+- **Job Management**: Provides a robust job management system that allows you to schedule and monitor document processing jobs.
+
+- **Integration**: Alfred can be easily integrated into your existing applications using its powerful API and SDKs.
+
+- **Scalability**: Alfred is designed to scale with your needs, whether you're processing thousands of documents a day or millions.
```
Comment (ypk46) [inline README.md:145]: I don't think this section should go at the end. I believe @rroa previous comments highlighted the features (bullet point) ordering (not the order in which this section appears within the whole README.md). The features list should effectively be ordered to reflect Alfred's file's journey in order to make it easier to catch.

**Review by ypk46 — CHANGES_REQUESTED** [review]
- I think some of the texts lack cohesion; they are missing commas, periods, and the correct connectors between ideas. All of my suggestions were generated and tuned using Chat GPT using your current text as input and adding the prompt `Improve this writing by fixing any grammar mistake, making it more cohesive and clear:`. Take advantage of using generative AI to improve you writing.

- I believe the main point is still missing from the discussion. It seems unnecessary to have a File section specifically. Instead, what I think is requested is a detailed walkthrough on how to use Alfred through the library. This would include a step-by-step guide from the perspective of a user (customer) on how to process a file using Alfred. Therefore, I suggest we adopt the customer's viewpoint and clearly outline the two main scenarios:

  - Uploading one or more remote files, which would immediately trigger a job.
  - Creating a session, uploading files in a deferred or asynchronous manner, and then creating the job.

  Both scenarios aim to use Alfred to process one or more files, albeit with different considerations.

File: README.md [hunk 7]
```diff
@@ -1,58 +1,237 @@
...
+   var uploadResult = await alfred.File.UploadAsync(uploadRequest);
+   ```
+
+3. Finally, handle the response accordingly. The response will contain the job ID since the `UploadAsync` method triggers job processing:</br></br>
+
+   ```csharp
+   Console.WriteLine(uploadResult.JobId);
+   ```
+
+**Complete example:**
```
Comment (ypk46) [inline README.md:142]: I don't think we need these `Complete example` section.

File: README.md [hunk 8]
```diff
@@ -1,58 +1,237 @@
...
+   **Example:**</br></br>
+
+   ```csharp
+   // Trigger job processing
+   Guid jobId = alfred.Job.CreateAsync(new CreateJobRequest { SessionId = sessionId}).Result.JobId;
+   ```
+
+**Complete example:**
+
+   ```csharp
```
Comment (ypk46) [inline README.md:228]: This is not even a "complete" example, is missing the actual file upload.

**Review by ypk46 — CHANGES_REQUESTED** [review]
This looks better, but we are still missing the querying for Job results.

File: README.md [hunk 9]
```diff
@@ -1,58 +1,237 @@
...
+
+- **Production**: The production environment are used for live applications and real-world scenarios. It is recommended to use this environment for production-ready applications. The frontend URL for the production environment is [https://app.tagshelf.com](https://app.tagshelf.com)</br></br>
+
+- **Staging**: The staging environment is used for testing and development purposes. It is recommended to use this environment for testing and development scenarios. The frontend URL for the staging environment is [https://staging.tagshelf.com](https://staging.tagshelf.com)</br></br>
+
+#### Authentication Methods
+
+The following examples demonstrate how to initialize the Alfred client with different authentication methods:
+
+   For OAuth authentication, specify the method and credentials explicitly and provide the environment type (Production or Staging):
```
Comment (MatehElismar) [inline README.md:67]: Where do I get my credentials from? By answering this question I think we cover this:

> 2. Where do I start so I can use this library that allows me to access Alfred?

File: README.md [hunk 10]
```diff
@@ -1,58 +1,237 @@
...
+
+   | Parameter | Type | Description |
+   | --- | --- | --- |
+   | JobId | Guid | Job ID |
+
+   **Example:**</br></br>
+
+   ```csharp
+   // Get job status
+   var jobStatus = await alfred.Job.GetAsync(uploadResult.JobId);
```
Comment (MatehElismar) [inline README.md:151]: > 4. How do I know its done?

After this, we could add information on job stages, what are the most important stages and which one determines that the process is done.

**Review by MatehElismar — CHANGES_REQUESTED** [review]

**Review by rroa — APPROVED** [review]
This is an improvement over the initial commit, but it still needs more work. I previously left some observations and guidelines in the form of questions for you to follow up on. There's no need to be specific by naming a "Step by Step guide"; I believe there are more elegant ways to achieve this.

I'm considering approving this now and then revisiting it for further improvements in our next sprint.

**Review by MatehElismar — APPROVED** [review]

**Review by ypk46 — APPROVED** [review]

**Comment by Edu-B** [comment]
@ypk46 thanks

**Pull request merged**